cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -1027,7 +1265,10 @@ predicate additionalLambdaFlowStep(Node nodeFrom, Node nodeTo, boolean preserves
  * One example would be to allow flow like `p.foo = p.bar;`, which is disallowed
  * by default as a heuristic.
  */
-predicate allowParameterReturnInSelf(ParameterNode p) { p instanceof IndirectParameterNode }
+predicate allowParameterReturnInSelf(ParameterNode p) {
+  p instanceof IndirectParameterNode
+  // TODO: Swift has a case for summarized callables here.
```

Comment:
And we should probably have that as well 😄 I think we can copy/paste the disjunct we have from Swift without any issues

Reply:
Good idea - just done it.  And yes, it was a copy-paste apart from correcting the name of one type.